### DISCUSSION TITLE
Implementing audio effect in raudio module

### DISCUSSION THREAD

**Discussion by adct-the-experimenter** [raysan5/raylib#1944] [Ideas]
I think it would be an interesting idea to implement audio effects into raudio module such as low-pass filter, high-pass filter, echo, reverb, etc.
This feature would be useful for making a special dynamic environment in a video game such as applying reverb to sounds in a cave level. 

Also it would be useful for a synth or other sound applications requiring effects.

I was wondering if others would be interested in working on this.

I am doing some work on it in my fork.
https://github.com/adct-the-experimenter/raylib

**Comment by raysan5** [comment]
@adct-the-experimenter `miniaudio` supports [multiple filters](https://miniaudio.docsforge.com/master/documentation/#filters) but they are not exposed by `raudio` at this moment. Maybe it can be considered for a future improvement, it requires investigating how they should be implemented.

**Reply by adct-the-experimenter** [reply]
Thank you for the response. I will look into how to implement those filters. 
I have an idea, but I need to test it.
